src/uu/tail/src/paths.rs

```diff
@@ -20,29 +23,44 @@ pub enum InputKind {
     Stdin,
 }
 
+#[cfg(unix)]
+impl From<&OsStr> for InputKind {
+    fn from(value: &OsStr) -> Self {
+        const DASH: [u8; 1] = [b'-'];
```

Comment:
I'm not sure, but I think this can be a `b"-"`?

Reply:
Yeah, but it's just a single character. It doesn't play a big role here but the result is an array `[u8; 1]` instead of a reference to an array `&[u8; 1]`.

Reply:
So you don't think 
```rust
const DASH: &[u8] = b"-";

if value.as_bytes() == DASH { }
```
is clearer?

Reply:
No not really, to me both look very similar and equally clear. However, the correct type of a byte string literal is `&[u8; n]` or better `&'static [u8; n]`. I first used `b"-"` but then changed it to `[b'-']` to spare me the reference here and I'm sure that `DASH` is stored on the stack. We're talking here about a single character. If you have a longer string literal and just need it as a (reference to a) byte array than you might spare yourself a lot of typing by using a byte string literal.

Reply:
Actually, I think I'd actually prefer this:
```rust
const DASH: &OsStr = OsStr::new("-");
if value == DASH { }
```
Then we don't even need to think about byte strings.

Reply:
`OsStr::new` can't be `const`. Why can't we just keep that little optimization the way it is and noone needs to look at any assembly code and rely on compiler magic?

Reply:
No, my priority here is clear code. I'd still push for the change even if it was technically slower, because it's not worth optimizing. The current code harder to read without any advantage.

Reply:
I think OsStr is the nicer here, it's clearer what's going on. `[b'-']` is hard to read.